docs/section5/Rule5-21.md

```diff
@@ -0,0 +1,66 @@
+
+  - Get matching building segment in P_RMR: `building_segment_p = match_data_element(P_RMR, BuildingSegments, building_segment_b.id)`
+    - Get total fenestration area for building segment in P_RMR: `total_fenestration_area_p = window_wall_areas_dictionary_p[building_segment_p.id][0]`
+
+  - For each thermal block in building segment: `for thermal_block_b in building_segment_b.thermal_blocks:`
+
+    - For each zone in thermal block: `for zone_b in thermal_block_b.zones:`  
+
+      - For each surface in zone: `for surface_b in zone_b.surfaces:`  
+
+        - Check if surface is above-grade wall and is regulated: `if ( get_opaque_surface_type(surface_b.id) == "ABOVE-GRADE WALL" ) AND ( scc_dictionary_b[surface_b.id] != "UNREGULATED" ):`
+
+          - For each subsurface in surface: `for subsurface_b in surface_b.subsurfaces:`
+
+            - Check if subsurface is door, and if the glazed vision area is more than 50% of the total area, add subsurface total area to surface total fenestration area: `if ( subsurface_b.classification == "DOOR" ) AND ( subsurface_b.glazed_area > ( subsurface_b.glazed_area + subsurface_b.opaque_area ) * 50% ): total_fenestration_area_surface_b += subsurface_b.glazed_area + subsurface_b.opaque_area`  
+
+            - Else if subsurface is not door, add total subsurface area to surface total fenestration area: `else if subsurface_b.classification != "DOOR" : total_fenestration_area_surface_b += subsurface_b.glazed_area + subsurface_b.opaque_area`
+
+          - Get matching surface in P_RMR: `surface_p = match_data_element(P_RMR, Surfaces, surface_b.id)`
+
+            - For each subsurface in surface: `for subsurface_p in surface_p.subsurfaces:`
+
+              - Check if subsurface is door, and if the glazed vision area is more than 50% of the total area, add subsurface total area to surface total fenestration area: `if ( subsurface_p.classification == "DOOR" ) AND ( subsurface_p.glazed_area > ( subsurface_p.glazed_area + subsurface_p.opaque_area ) * 50% ): total_fenestration_area_surface_p += subsurface_p.glazed_area + subsurface_p.opaque_area`  
+
+              - Else if subsurface is not door, add total subsurface area to surface total fenestration area: `else if subsurface_p.classification != "DOOR" : total_fenestration_area_surface_p += surface_p.glazed_area + surface_p.opaque_area`
+
+        **Rule Assertion:**
+
+        - Case 1: For each surface, if total fenestration area in B_RMR is in the same proportion as in P_RMR: `if total_fenestration_area_surface_b / total_fenestration_area_b == total_fenestration_area_surface_p / total_fenestration_area_p: PASS`
+
+        - Case 2: Else: `else: FAIL and raise_warning "THE DISTRIBUTION OF VERTICAL FENESTRATION IN SURFACE IS NOT IN THE SAME PROPORTION AS IN THE PROPOSED DESIGN. VERIFY IF ENVELOPE IS EXISTING OR ALTERED AND CAN BE EXCLUDED FROM THIS CHECK."`
+
```

Comment:
Please change "raise_warning" to "show message". Please make it say ""THE VERTICAL FENESTRATION IS NOT DISRIBUTED ACROSS BASELINE OPAQUE SURFACES IN THE SAME PROPORTION AS IN THE PROPOSED DESIGN. VERIFY IF ENVELOPE IS EXISTING OR ALTERED AND CAN BE EXCLUDED FROM THIS CHECK."

Reply:
Updated.